`dockerbuild` does not work on Windows

```
plugins {
    id("org.jetbrains.kotlin.jvm") version "1.4.30"
    id("org.jetbrains.kotlin.kapt") version "1.4.30"
    id("org.jetbrains.kotlin.plugin.allopen") version "1.4.30"
    id("com.github.johnrengelman.shadow") version "6.1.0"
    id("io.micronaut.application") version "1.3.4"
}
```

I am on Windows 10.

`./gradlew dockerBuild` gets me this stacktrace. Googling the stack gives bmuschko/gradle-docker-plugin#432, but that should've already been fixed.

```
2:12:37 PM: Executing task 'dockerBuild --stacktrace'...

Starting Gradle Daemon...
Gradle Daemon started in 2 s 241 ms
> Task :kaptGenerateStubsKotlin
> Task :processResources

> Task :dockerfile
Dockerfile written to: C:\Users\quydx\IdeaProjects\ln-or-summary\build\docker\Dockerfile

> Task :kaptKotlin
Note: Creating bean classes for 1 type elements
> Task :compileKotlin
> Task :compileJava NO-SOURCE
> Task :classes
> Task :runnerJar
> Task :buildLayers

> Task :dockerBuild FAILED
Building image using context 'C:\Users\quydx\IdeaProjects\ln-or-summary'.
Using Dockerfile 'C:\Users\quydx\IdeaProjects\ln-or-summary\build\docker\Dockerfile'

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dockerBuild'.
> Error occurred while preparing Docker context folder.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':dockerBuild'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$3(ExecuteActionsTaskExecuter.java:186)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:268)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:184)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:173)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:411)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:398)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:391)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:377)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: com.github.dockerjava.api.exception.DockerClientException: Error occurred while preparing Docker context folder.
	at com.github.dockerjava.core.dockerfile.Dockerfile$ScannedResult.buildDockerFolderTar(Dockerfile.java:166)
	at com.github.dockerjava.core.dockerfile.Dockerfile$ScannedResult.buildDockerFolderTar(Dockerfile.java:124)
	at com.github.dockerjava.core.command.BuildImageCmdImpl.withDockerfile(BuildImageCmdImpl.java:346)
	at com.github.dockerjava.core.command.BuildImageCmdImpl.withDockerfile(BuildImageCmdImpl.java:23)
	at com.bmuschko.gradle.docker.tasks.image.DockerBuildImage.runRemoteCommand(DockerBuildImage.groovy:235)
	at com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask.start(AbstractDockerRemoteApiTask.groovy:73)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$2.run(ExecuteActionsTaskExecuter.java:494)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:56)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$run$1(DefaultBuildOperationExecutor.java:71)
	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.runWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:45)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:71)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:479)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:462)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$400(ExecuteActionsTaskExecuter.java:105)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:273)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:251)
	at org.gradle.internal.execution.steps.ExecuteStep.lambda$executeOperation$1(ExecuteStep.java:66)
	at org.gradle.internal.execution.steps.ExecuteStep.executeOperation(ExecuteStep.java:66)
	at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:34)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:47)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:44)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:44)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:34)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:72)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:42)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:53)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:39)
	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:44)
	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:77)
	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:58)
	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:54)
	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:32)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:57)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:38)
	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:63)
	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:30)
	at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:176)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:76)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:47)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:43)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:32)
	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:39)
	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:25)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:102)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:95)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:83)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:44)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:96)
	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:52)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:83)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:54)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:88)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:88)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:46)
	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:34)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:43)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution$3.withWorkspace(ExecuteActionsTaskExecuter.java:286)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:43)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:33)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:40)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:30)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:54)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:40)
	at org.gradle.internal.execution.impl.DefaultExecutionEngine.rebuild(DefaultExecutionEngine.java:46)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$0(ExecuteActionsTaskExecuter.java:182)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:182)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:173)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:411)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:398)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:391)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:377)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: java.io.IOException: The process cannot access the file because another process has locked a portion of the file
	at com.bmuschko.gradle.docker.shaded.com.google.common.io.ByteStreams.copy(ByteStreams.java:110)
	at com.github.dockerjava.core.util.CompressArchiveUtil.addFileToTar(CompressArchiveUtil.java:44)
	at com.github.dockerjava.core.util.CompressArchiveUtil.archiveTARFiles(CompressArchiveUtil.java:106)
	at com.github.dockerjava.core.dockerfile.Dockerfile$ScannedResult.buildDockerFolderTar(Dockerfile.java:134)
	... 128 more
	Suppressed: java.io.IOException: This archive contains unclosed entries.
		at com.bmuschko.gradle.docker.shaded.org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.finish(TarArchiveOutputStream.java:289)
		at com.bmuschko.gradle.docker.shaded.org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.close(TarArchiveOutputStream.java:307)
		at com.github.dockerjava.core.util.CompressArchiveUtil.archiveTARFiles(CompressArchiveUtil.java:108)
		at com.github.dockerjava.core.dockerfile.Dockerfile$ScannedResult.buildDockerFolderTar(Dockerfile.java:134)
		at com.github.dockerjava.core.dockerfile.Dockerfile$ScannedResult.buildDockerFolderTar(Dockerfile.java:124)
		at com.github.dockerjava.core.command.BuildImageCmdImpl.withDockerfile(BuildImageCmdImpl.java:346)
		at com.github.dockerjava.core.command.BuildImageCmdImpl.withDockerfile(BuildImageCmdImpl.java:23)
		at com.bmuschko.gradle.docker.tasks.image.DockerBuildImage.runRemoteCommand(DockerBuildImage.groovy:235)
		at com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask.start(AbstractDockerRemoteApiTask.groovy:73)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
		at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
		at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
		at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$2.run(ExecuteActionsTaskExecuter.java:494)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
		at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
		at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
		at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:56)
		at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$run$1(DefaultBuildOperationExecutor.java:71)
		at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.runWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:45)
		at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:71)
		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:479)
		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:462)
		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$400(ExecuteActionsTaskExecuter.java:105)
		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:273)
		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:251)
		at org.gradle.internal.execution.steps.ExecuteStep.lambda$executeOperation$1(ExecuteStep.java:66)
		at java.base/java.util.Optional.orElseGet(Optional.java:369)
		at org.gradle.internal.execution.steps.ExecuteStep.executeOperation(ExecuteStep.java:66)
		at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:34)
		at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:47)
		at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:44)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
		at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
		at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
		at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
		at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
		at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
		at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
		at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:44)
		at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:34)
		at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:72)
		at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:42)
		at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:53)
		at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:39)
		at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:44)
		at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:77)
		at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:58)
		at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:54)
		at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:32)
		at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:57)
		at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:38)
		at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:63)
		at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:30)
		at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:176)
		at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:76)
		at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:47)
		at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:43)
		at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:32)
		at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:39)
		at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:25)
		at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:102)
		at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:95)
		at java.base/java.util.Optional.map(Optional.java:265)
		at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
		at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
		at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:83)
		at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:44)
		at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
		at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
		at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:96)
		at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:52)
		at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:83)
		at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:54)
		at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
		at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:88)
		at java.base/java.util.Optional.orElseGet(Optional.java:369)
		at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:88)
		at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
		at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
		at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:46)
		at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:34)
		at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:43)
		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution$3.withWorkspace(ExecuteActionsTaskExecuter.java:286)
		at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:43)
		at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:33)
		at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:40)
		at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:30)
		at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:54)
		at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:40)
		at org.gradle.internal.execution.impl.DefaultExecutionEngine.rebuild(DefaultExecutionEngine.java:46)
		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$0(ExecuteActionsTaskExecuter.java:182)
		at java.base/java.util.Optional.map(Optional.java:265)
		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:182)
		at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:173)
		at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
		at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
		at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
		at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
		at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
		at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
		at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
		at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
		at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
		at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
		at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
		at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
		at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
		at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
		at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
		at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
		at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
		at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
		at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:411)
		at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:398)
		at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:391)
		at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:377)
		at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
		at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
		at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
		at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
		at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
		at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
		at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
		at java.base/java.lang.Thread.run(Thread.java:834)


* Get more help at https://help.gradle.org

BUILD FAILED in 17s
8 actionable tasks: 8 executed
2:12:57 PM: Task execution finished 'dockerBuild --stacktrace'.
```

--- a/.github/workflows/graalvm.yml
+++ b/.github/workflows/graalvm.yml
@@ -14,7 +14,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        graalvm: ['21.0.0.java8', '21.0.0.java11']
+        graalvm: ['21.0.0.2.java8', '21.0.0.2.java11']
     steps:
       - uses: actions/checkout@v2
       - uses: actions/cache@v2.1.4
@@ -24,7 +24,7 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-gradle-
       - name: Setup GraalVM CE
-        uses: DeLaGuardo/setup-graalvm@8bbfe44ef9c6f5c07e5af036a1bffd561c037d18
+        uses: DeLaGuardo/setup-graalvm@4.0
         with:
           graalvm-version: ${{ matrix.graalvm }}
       - name: Install Native Image

--- a/.github/workflows/windows-ci.yml
+++ b/.github/workflows/windows-ci.yml
@@ -28,9 +28,11 @@ jobs:
       - name: Install GraalVM
         run: Expand-Archive -path 'graal.zip' -destinationpath '.'
       - name: Install Native Image
-        run: graalvm-ce-java11-21.0.0.2\bin\gu.cmd install native-image
+        run: graalvm-ce-java11-${{ matrix.graalvm }}\bin\gu.cmd install native-image
       - name: Build with Gradle
         run: ./gradlew.bat check --no-daemon --continue
         env:
+          JAVA_HOME: graalvm-ce-java11-${{ matrix.graalvm }}\bin
+          GRAALVM_HOME: graalvm-ce-java11-${{ matrix.graalvm }}\bin
           TESTCONTAINERS_RYUK_DISABLED: true
 

--- a/.github/workflows/windows-ci.yml
+++ b/.github/workflows/windows-ci.yml
@@ -11,7 +11,7 @@ on:
 jobs:
   build:
     if: github.repository != 'micronaut-projects/micronaut-project-template'
-    runs-on: windows-latest
+    runs-on: windows-2016
     strategy:
       matrix:
         graalvm: ['21.0.0.2']
@@ -29,8 +29,14 @@ jobs:
         run: Expand-Archive -path 'graal.zip' -destinationpath '.'
       - name: Install Native Image
         run: graalvm-ce-java11-${{ matrix.graalvm }}\bin\gu.cmd install native-image
+      - name: Set up Visual C Build Tools Workload for Visual Studio 2017 Build Tools
+        run: |
+          choco install visualstudio2017-workload-vctools
       - name: Build with Gradle
-        run: ./gradlew.bat check --no-daemon --continue
+        shell: cmd
+        run: |
+          call "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build\vcvars64.bat"
+          ./gradlew.bat check --no-daemon --continue
         env:
           JAVA_HOME: graalvm-ce-java11-${{ matrix.graalvm }}
           GRAALVM_HOME: graalvm-ce-java11-${{ matrix.graalvm }}

--- a/.github/workflows/windows-ci.yml
+++ b/.github/workflows/windows-ci.yml
@@ -26,9 +26,9 @@ jobs:
       - name: Download GraalVM
         run: Invoke-RestMethod -Uri https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-${{ matrix.graalvm }}/graalvm-ce-java11-windows-amd64-${{ matrix.graalvm }}.zip -OutFile 'graal.zip'
       - name: Install GraalVM
-        run: Expand-Archive -path 'graal.zip' -destinationpath '.'
+        run: Expand-Archive -path 'graal.zip' -destinationpath 'C:\graalvm'
       - name: Install Native Image
-        run: graalvm-ce-java11-${{ matrix.graalvm }}\bin\gu.cmd install native-image
+        run: C:\graalvm\graalvm-ce-java11-${{ matrix.graalvm }}\bin\gu.cmd install native-image
       - name: Set up Visual C Build Tools Workload for Visual Studio 2017 Build Tools
         run: |
           choco install visualstudio2017-workload-vctools
@@ -38,7 +38,7 @@ jobs:
           call "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build\vcvars64.bat"
           ./gradlew.bat check --no-daemon --continue
         env:
-          JAVA_HOME: graalvm-ce-java11-${{ matrix.graalvm }}
-          GRAALVM_HOME: graalvm-ce-java11-${{ matrix.graalvm }}
+          JAVA_HOME: C:\graalvm\graalvm-ce-java11-${{ matrix.graalvm }}
+          GRAALVM_HOME: C:\graalvm\graalvm-ce-java11-${{ matrix.graalvm }}
           TESTCONTAINERS_RYUK_DISABLED: true
 

--- a/.github/workflows/windows-ci.yml
+++ b/.github/workflows/windows-ci.yml
@@ -29,6 +29,8 @@ jobs:
         run: Expand-Archive -path 'graal.zip' -destinationpath 'C:\graalvm'
       - name: Install Native Image
         run: C:\graalvm\graalvm-ce-java11-${{ matrix.graalvm }}\bin\gu.cmd install native-image
+      - name: Verify Native Image
+        run: C:\graalvm\graalvm-ce-java11-${{ matrix.graalvm }}\bin\native-image --version
       - name: Set up Visual C Build Tools Workload for Visual Studio 2017 Build Tools
         run: |
           choco install visualstudio2017-workload-vctools

--- a/.github/workflows/windows-ci.yml
+++ b/.github/workflows/windows-ci.yml
@@ -32,7 +32,7 @@ jobs:
       - name: Build with Gradle
         run: ./gradlew.bat check --no-daemon --continue
         env:
-          JAVA_HOME: graalvm-ce-java11-${{ matrix.graalvm }}\bin
-          GRAALVM_HOME: graalvm-ce-java11-${{ matrix.graalvm }}\bin
+          JAVA_HOME: graalvm-ce-java11-${{ matrix.graalvm }}
+          GRAALVM_HOME: graalvm-ce-java11-${{ matrix.graalvm }}
           TESTCONTAINERS_RYUK_DISABLED: true
 

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerPlugin.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerPlugin.java
@@ -187,6 +187,7 @@ public class MicronautDockerPlugin implements Plugin<Project> {
                 task.getDockerFile()
                         .convention(dockerFileTask.flatMap(Dockerfile::getDestFile));
             }
+            task.getInputDir().set(project.getBuildDir());
             task.getImages().set(Collections.singletonList(project.getName()));
         });
 
@@ -222,7 +223,7 @@ public class MicronautDockerPlugin implements Plugin<Project> {
         dockerBuildTask.configure(task -> {
             task.setGroup(BasePlugin.BUILD_GROUP);
             task.setDescription("Builds a Native Docker Image using GraalVM");
-            task.getInputDir().set(project.getProjectDir());
+            task.getInputDir().set(project.getBuildDir());
             if (f.exists()) {
                 task.getDockerFile().set(f);
             } else {

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerPlugin.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerPlugin.java
@@ -35,7 +35,6 @@ public class MicronautDockerPlugin implements Plugin<Project> {
         TaskContainer tasks = project.getTasks();
         ExtensionContainer extensions = project.getExtensions();
         extensions.create("docker", DockerExtension.class);
-        File applicationLayout = new File(project.getBuildDir(), "layers");
         TaskProvider<Jar> runnerJar = tasks.register("runnerJar", Jar.class, jar -> {
             jar.dependsOn(tasks.findByName("classes"));
             jar.getArchiveClassifier().set("runner");
@@ -83,7 +82,11 @@ public class MicronautDockerPlugin implements Plugin<Project> {
         });
         TaskProvider<Task> buildLayersTask = tasks.register("buildLayers", task -> {
             task.dependsOn(runnerJar);
+            task.setGroup(BasePlugin.BUILD_GROUP);
+            task.setDescription("Builds application layers for use in a Docker container");
 
+        });
+        buildLayersTask.configure((task -> {
             Configuration runtimeClasspath = project.getConfigurations()
                     .getByName(RUNTIME_CLASSPATH_CONFIGURATION_NAME);
             TaskOutputs jar = runnerJar.get().getOutputs();
@@ -101,10 +104,8 @@ public class MicronautDockerPlugin implements Plugin<Project> {
             inputs.files(jarFiles);
             inputs.files(resourcesDir);
 
-
+            File applicationLayout = new File(project.getBuildDir(), "layers");
             task.getOutputs().dir(applicationLayout);
-            task.setGroup(BasePlugin.BUILD_GROUP);
-            task.setDescription("Builds application layers for use in a Docker container");
 
             // NOTE: Has to be an anonymous inner class otherwise incremental build does not work
             // DO NOT REPLACE WITH LAMBDA
@@ -128,7 +129,7 @@ public class MicronautDockerPlugin implements Plugin<Project> {
                     );
                 }
             });
-        });
+        }));
 
         Task assemble = tasks.findByName("assemble");
         if (assemble != null) {

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
@@ -1,7 +1,6 @@
 package io.micronaut.gradle.docker;
 
 import com.bmuschko.gradle.docker.tasks.image.Dockerfile;
-import org.gradle.api.GradleException;
 import org.gradle.api.JavaVersion;
 import org.gradle.api.Project;
 import org.gradle.api.model.ObjectFactory;
@@ -12,10 +11,6 @@ import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.Input;
 import org.gradle.internal.jvm.Jvm;
 
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
@@ -47,30 +42,6 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
         this.args = objects.listProperty(String.class);
         this.exposedPorts = objects.listProperty(Integer.class)
                     .convention(Collections.singletonList(8080));
-        final java.io.File dockerFile;
-        try {
-            final java.io.File buildDir = project.getBuildDir().getCanonicalFile();
-            System.out.println("File pathSeparator = " + java.io.File.pathSeparator);
-            System.out.println("File pathSeparator len = " + java.io.File.pathSeparator.length());
-            System.out.println("File separator = " + java.io.File.separator);
-            System.out.println("File separator len = " + java.io.File.separator.length());
-            System.out.println("buildDir = " + buildDir);
-            final Path path = buildDir.toPath();
-            System.out.println("Path = " + path);
-            System.out.println("path.getFileSystem() = " + path.getFileSystem());
-            System.out.println("path.toUri() = " + path.toUri());
-            dockerFile = new java.io.File(buildDir, "docker/Dockerfile");
-        } catch (IOException e) {
-            e.printStackTrace();
-            throw new GradleException("Invalid Build directory: " + project.getBuildDir() + ": " + e.getMessage());
-        }
-        try {
-            final java.io.File canonicalFile = dockerFile.getCanonicalFile();
-            System.out.println("canonicalFile = " + canonicalFile);
-            this.getDestFile().set(canonicalFile);
-        } catch (IOException e) {
-            throw new GradleException("Invalid Target Dockerfile: " + dockerFile);
-        }
 
         doLast(task -> {
             java.io.File f = getDestFile().get().getAsFile();
@@ -95,9 +66,9 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
                 javaApplication.getMainClass().set("com.fnproject.fn.runtime.EntryPoint");
                 from(new Dockerfile.From(from != null ? from : "fnproject/fn-java-fdk:" + getProjectFnVersion()));
                 workingDir("/function");
-                copyFile("build/layers/libs/*.jar", "/function/app/");
-                copyFile("build/layers/resources/*", "/function/app/");
-                copyFile("build/layers/application.jar", "/function/app/");
+                copyFile("layers/libs/*.jar", "/function/app/");
+                copyFile("layers/resources/*", "/function/app/");
+                copyFile("layers/application.jar", "/function/app/");
                 String cmd = this.defaultCommand.get();
                 if ("none".equals(cmd)) {
                     super.defaultCommand("io.micronaut.oraclecloud.function.http.HttpFunction::handleRequest");
@@ -200,8 +171,8 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
 
     static void setupResources(Dockerfile task) {
         task.workingDir("/home/app");
-        task.copyFile("build/layers/libs", "/home/app/libs");
-        task.copyFile("build/layers/resources", "/home/app/resources");
-        task.copyFile("build/layers/application.jar", "/home/app/application.jar");
+        task.copyFile("layers/libs", "/home/app/libs");
+        task.copyFile("layers/resources", "/home/app/resources");
+        task.copyFile("layers/application.jar", "/home/app/application.jar");
     }
 }

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
@@ -79,7 +79,7 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
             case LAMBDA:
                 javaApplication.getMainClass().set("io.micronaut.function.aws.runtime.MicronautLambdaRuntime");
             default:
-                from(new Dockerfile.From(from != null ? from : "openjdk:15-alpine"));
+                from(new Dockerfile.From(from != null ? from : "openjdk:16-alpine"));
                 setupResources(this);
                 exposePort(exposedPorts);
                 entryPoint(getArgs().map(strings -> {

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
@@ -59,7 +59,8 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
             System.out.println("path.toUri() = " + path.toUri());
             dockerFile = new java.io.File(buildDir, "docker/Dockerfile");
         } catch (IOException e) {
-            throw new GradleException("Invalid Build directory: " + project.getBuildDir());
+            e.printStackTrace();
+            throw new GradleException("Invalid Build directory: " + project.getBuildDir() + ": " + e.getMessage());
         }
         try {
             final java.io.File canonicalFile = dockerFile.getCanonicalFile();

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
@@ -55,7 +55,7 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
             System.out.println("File separator = " + java.io.File.separator);
             System.out.println("File separator len = " + java.io.File.separator.length());
             System.out.println("buildDir = " + buildDir);
-            final Path path = buildDir.toPath().toRealPath();
+            final Path path = buildDir.toPath();
             System.out.println("Path = " + path);
             System.out.println("path.getFileSystem() = " + path.getFileSystem());
             System.out.println("path.toUri() = " + path.toUri());

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
@@ -45,7 +45,14 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
         this.args = objects.listProperty(String.class);
         this.exposedPorts = objects.listProperty(Integer.class)
                     .convention(Collections.singletonList(8080));
-        final java.io.File dockerFile = new java.io.File(project.getBuildFile(), "/docker/Dockerfile");
+        final java.io.File dockerFile;
+        try {
+            final java.io.File buildDir = project.getBuildDir().getCanonicalFile();
+            System.out.println("buildDir = " + buildDir);
+            dockerFile = new java.io.File(buildDir, "docker/Dockerfile");
+        } catch (IOException e) {
+            throw new GradleException("Invalid Build directory: " + project.getBuildDir());
+        }
         try {
             final java.io.File canonicalFile = dockerFile.getCanonicalFile();
             System.out.println("canonicalFile = " + canonicalFile);

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
@@ -14,6 +14,8 @@ import org.gradle.internal.jvm.Jvm;
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
@@ -51,6 +53,10 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
             System.out.println("File pathSeparator = " + java.io.File.pathSeparator);
             System.out.println("File separator = " + java.io.File.separator);
             System.out.println("buildDir = " + buildDir);
+            final Path path = buildDir.toPath().toRealPath();
+            System.out.println("Path = " + path);
+            System.out.println("path.getFileSystem() = " + path.getFileSystem());
+            System.out.println("path.toUri() = " + path.toUri());
             dockerFile = new java.io.File(buildDir, "docker/Dockerfile");
         } catch (IOException e) {
             throw new GradleException("Invalid Build directory: " + project.getBuildDir());

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
@@ -51,7 +51,9 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
         try {
             final java.io.File buildDir = project.getBuildDir().getCanonicalFile();
             System.out.println("File pathSeparator = " + java.io.File.pathSeparator);
+            System.out.println("File pathSeparator len = " + java.io.File.pathSeparator.length());
             System.out.println("File separator = " + java.io.File.separator);
+            System.out.println("File separator len = " + java.io.File.separator.length());
             System.out.println("buildDir = " + buildDir);
             final Path path = buildDir.toPath().toRealPath();
             System.out.println("Path = " + path);

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
@@ -1,6 +1,7 @@
 package io.micronaut.gradle.docker;
 
 import com.bmuschko.gradle.docker.tasks.image.Dockerfile;
+import org.gradle.api.GradleException;
 import org.gradle.api.JavaVersion;
 import org.gradle.api.Project;
 import org.gradle.api.model.ObjectFactory;
@@ -11,6 +12,8 @@ import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.Input;
 import org.gradle.internal.jvm.Jvm;
 
+import java.io.File;
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
@@ -42,6 +45,14 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
         this.args = objects.listProperty(String.class);
         this.exposedPorts = objects.listProperty(Integer.class)
                     .convention(Collections.singletonList(8080));
+        final java.io.File dockerFile = new java.io.File(project.getBuildFile(), "/docker/Dockerfile");
+        try {
+            final java.io.File canonicalFile = dockerFile.getCanonicalFile();
+            System.out.println("canonicalFile = " + canonicalFile);
+            this.getDestFile().set(canonicalFile);
+        } catch (IOException e) {
+            throw new GradleException("Invalid Target Dockerfile: " + dockerFile);
+        }
 
         doLast(task -> {
             java.io.File f = getDestFile().get().getAsFile();

--- a/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/MicronautDockerfile.java
@@ -48,6 +48,8 @@ public class MicronautDockerfile extends Dockerfile implements DockerBuildOption
         final java.io.File dockerFile;
         try {
             final java.io.File buildDir = project.getBuildDir().getCanonicalFile();
+            System.out.println("File pathSeparator = " + java.io.File.pathSeparator);
+            System.out.println("File separator = " + java.io.File.separator);
             System.out.println("buildDir = " + buildDir);
             dockerFile = new java.io.File(buildDir, "docker/Dockerfile");
         } catch (IOException e) {

--- a/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
@@ -64,10 +64,15 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
     private String findNativeImage(String... envs) {
         for (String env : envs) {
             final String graalvmHome = System.getenv(env);
+            System.out.println("env = " + env);
+            System.out.println("graalvmHome = " + graalvmHome);
             if (graalvmHome != null && graalvmHome.length() > 0) {
                 final File ni = new File(graalvmHome, "bin/native-image");
+                System.out.println("Checking Native Image = " + ni);
                 if (ni.exists()) {
                     return ni.getAbsolutePath();
+                } else {
+                    System.out.println("Doesn't exist!");
                 }
             }
         }

--- a/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
@@ -62,15 +62,27 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
     }
 
     private String findNativeImage(String graalHome, String javaHome) {
-        if (graalHome != null && graalHome.length() > 0) {
-            return new File(graalHome, "bin/native-image").getAbsolutePath();
-        } else if (javaHome != null && javaHome.length() > 0) {
-            File ni = new File(javaHome, "bin/native-image");
-            if (ni.exists()) {
-                return ni.getAbsolutePath();
+        if (isNotBlank(graalHome)) {
+            graalHome = System.getenv(graalHome);
+            if (isNotBlank(graalHome)) {
+                return new File(graalHome, "bin/native-image").getAbsolutePath();
+            }
+        }
+
+        if (isNotBlank(javaHome)) {
+            javaHome = System.getenv(javaHome);
+            if (isNotBlank(javaHome)) {
+                File ni = new File(javaHome, "bin/native-image");
+                if (ni.exists()) {
+                    return ni.getAbsolutePath();
+                }
             }
         }
         return "native-image";
+    }
+
+    private boolean isNotBlank(String graalHome) {
+        return graalHome != null && graalHome.length() > 0;
     }
 
     @Override

--- a/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
@@ -71,7 +71,9 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
             if (isNotBlank(graalHome)) {
                 try {
                     final File f = getNativeImageExecutable(graalHome);
-                    return f.getCanonicalPath();
+                    if (f.exists()) {
+                        return f.getCanonicalPath();
+                    }
                 } catch (IOException e) {
                     // continue
                 }
@@ -99,10 +101,19 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
         final File f;
         if (Os.isFamily("windows")) {
             f = new File(javaHome, "bin/native-image.exe");
+            if (f.exists()) {
+                return f;
+            } else {
+                return new File(javaHome, "lib/svm/bin/native-image.exe");
+            }
         } else {
             f = new File(javaHome, "bin/native-image");
+            if (f.exists()) {
+                return f;
+            } else {
+                return new File(javaHome, "lib/svm/bin/native-image");
+            }
         }
-        return f;
     }
 
     private boolean isNotBlank(String graalHome) {

--- a/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
@@ -61,24 +61,13 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
         booleanCmds.put(isVerbose::get,  "--verbose");
     }
 
-    private String findNativeImage(String... envs) {
-        for (String env : envs) {
-            final String graalvmHome = System.getenv(env);
-            System.out.println("env = " + env);
-            System.out.println("graalvmHome = " + graalvmHome);
-            if (graalvmHome != null && graalvmHome.length() > 0) {
-                File ni = new File(graalvmHome, "bin/native-image");
-                System.out.println("Checking Native Image = " + ni);
-                if (ni.exists()) {
-                    return ni.getAbsolutePath();
-                } else {
-                    ni = new File(graalvmHome, "bin/native-image.exe");
-                    if (ni.exists()) {
-                        return ni.getAbsolutePath();
-                    } else {
-                        System.out.println("Doesn't exist!");
-                    }
-                }
+    private String findNativeImage(String graalHome, String javaHome) {
+        if (graalHome != null && graalHome.length() > 0) {
+            return new File(graalHome, "bin/native-image").getAbsolutePath();
+        } else if (javaHome != null && javaHome.length() > 0) {
+            File ni = new File(javaHome, "bin/native-image");
+            if (ni.exists()) {
+                return ni.getAbsolutePath();
             }
         }
         return "native-image";

--- a/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
@@ -40,7 +40,6 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
         super(NativeImageTask.class);
         String nativeImageExecutable = findNativeImage("GRAALVM_HOME", "JAVA_HOME");
 
-        System.out.println("nativeImageExecutable = " + nativeImageExecutable);
         setExecutable(nativeImageExecutable);
         setWorkingDir(new File(getProject().getBuildDir(), "native-image"));
         ObjectFactory objectFactory = getObjectFactory();

--- a/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
@@ -12,6 +12,7 @@ import org.gradle.api.tasks.*;
 import javax.annotation.Nullable;
 import javax.inject.Inject;
 import java.io.File;
+import java.io.IOException;
 import java.util.*;
 import java.util.function.BooleanSupplier;
 
@@ -39,7 +40,9 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
     public NativeImageTask() {
         super(NativeImageTask.class);
         String nativeImageExecutable = findNativeImage("GRAALVM_HOME", "JAVA_HOME");
-
+        if (!new File(nativeImageExecutable).exists()) {
+            System.out.println("DOESN'T EXIST!! = " + nativeImageExecutable);
+        }
         setExecutable(nativeImageExecutable);
         setWorkingDir(new File(getProject().getBuildDir(), "native-image"));
         ObjectFactory objectFactory = getObjectFactory();
@@ -64,7 +67,11 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
         if (isNotBlank(graalHome)) {
             graalHome = System.getenv(graalHome);
             if (isNotBlank(graalHome)) {
-                return new File(graalHome, "bin/native-image").getAbsolutePath();
+                try {
+                    return new File(graalHome, "bin/native-image").getCanonicalPath();
+                } catch (IOException e) {
+                    // continue
+                }
             }
         }
 
@@ -73,7 +80,11 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
             if (isNotBlank(javaHome)) {
                 File ni = new File(javaHome, "bin/native-image");
                 if (ni.exists()) {
-                    return ni.getAbsolutePath();
+                    try {
+                        return ni.getCanonicalPath();
+                    } catch (IOException e) {
+                        // continue
+                    }
                 }
             }
         }

--- a/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
@@ -39,6 +39,8 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
     public NativeImageTask() {
         super(NativeImageTask.class);
         String nativeImageExecutable = findNativeImage("GRAALVM_HOME", "JAVA_HOME");
+
+        System.out.println("nativeImageExecutable = " + nativeImageExecutable);
         setExecutable(nativeImageExecutable);
         setWorkingDir(new File(getProject().getBuildDir(), "native-image"));
         ObjectFactory objectFactory = getObjectFactory();

--- a/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
@@ -67,12 +67,17 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
             System.out.println("env = " + env);
             System.out.println("graalvmHome = " + graalvmHome);
             if (graalvmHome != null && graalvmHome.length() > 0) {
-                final File ni = new File(graalvmHome, "bin/native-image");
+                File ni = new File(graalvmHome, "bin/native-image");
                 System.out.println("Checking Native Image = " + ni);
                 if (ni.exists()) {
                     return ni.getAbsolutePath();
                 } else {
-                    System.out.println("Doesn't exist!");
+                    ni = new File(graalvmHome, "bin/native-image.exe");
+                    if (ni.exists()) {
+                        return ni.getAbsolutePath();
+                    } else {
+                        System.out.println("Doesn't exist!");
+                    }
                 }
             }
         }

--- a/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/NativeImageTask.java
@@ -1,5 +1,6 @@
 package io.micronaut.gradle.graalvm;
 
+import org.apache.tools.ant.taskdefs.condition.Os;
 import org.gradle.api.Project;
 import org.gradle.api.file.FileCollection;
 import org.gradle.api.model.ObjectFactory;
@@ -8,6 +9,7 @@ import org.gradle.api.provider.ListProperty;
 import org.gradle.api.provider.MapProperty;
 import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.*;
+import org.jetbrains.annotations.NotNull;
 
 import javax.annotation.Nullable;
 import javax.inject.Inject;
@@ -68,7 +70,8 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
             graalHome = System.getenv(graalHome);
             if (isNotBlank(graalHome)) {
                 try {
-                    return new File(graalHome, "bin/native-image").getCanonicalPath();
+                    final File f = getNativeImageExecutable(graalHome);
+                    return f.getCanonicalPath();
                 } catch (IOException e) {
                     // continue
                 }
@@ -78,10 +81,10 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
         if (isNotBlank(javaHome)) {
             javaHome = System.getenv(javaHome);
             if (isNotBlank(javaHome)) {
-                File ni = new File(javaHome, "bin/native-image");
-                if (ni.exists()) {
+                final File f = getNativeImageExecutable(javaHome);
+                if (f.exists()) {
                     try {
-                        return ni.getCanonicalPath();
+                        return f.getCanonicalPath();
                     } catch (IOException e) {
                         // continue
                     }
@@ -89,6 +92,17 @@ public class NativeImageTask extends AbstractExecTask<NativeImageTask>
             }
         }
         return "native-image";
+    }
+
+    @NotNull
+    private File getNativeImageExecutable(String javaHome) {
+        final File f;
+        if (Os.isFamily("windows")) {
+            f = new File(javaHome, "bin/native-image.exe");
+        } else {
+            f = new File(javaHome, "bin/native-image");
+        }
+        return f;
     }
 
     private boolean isNotBlank(String graalHome) {

--- a/src/test/groovy/io/micronaut/gradle/BuildLayersSpec.groovy
+++ b/src/test/groovy/io/micronaut/gradle/BuildLayersSpec.groovy
@@ -1,0 +1,62 @@
+package io.micronaut.gradle
+
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+
+class BuildLayersSpec extends Specification {
+
+    @Rule TemporaryFolder testProjectDir = new TemporaryFolder()
+
+    File settingsFile
+    File buildFile
+
+    def setup() {
+        settingsFile = testProjectDir.newFile('settings.gradle')
+        buildFile = testProjectDir.newFile('build.gradle')
+    }
+
+    void 'test build layers with custom build directory'() {
+        given:
+        settingsFile << "rootProject.name = 'hello-world'"
+        buildFile << """
+            plugins {
+                id "io.micronaut.application"
+            }
+            
+            micronaut {
+                version "2.4.2"
+                runtime "netty"
+                testRuntime "junit5"
+            }
+            
+            repositories {
+                mavenCentral()
+            }
+            
+            dependencies {
+                runtimeOnly("ch.qos.logback:logback-classic")
+                testImplementation("io.micronaut:micronaut-http-client")
+            }
+            mainClassName="example.Application"
+            
+            project.buildDir = "build-custom"
+        """
+
+        when:
+        def result = GradleRunner.create()
+                .withProjectDir(testProjectDir.root)
+                .withArguments('buildLayers')
+                .withPluginClasspath()
+                .build()
+
+        def task = result.task(":buildLayers")
+
+        then:
+        task.outcome == TaskOutcome.SUCCESS
+        new File(testProjectDir.root, "build-custom/layers").exists()
+        !new File(testProjectDir.root, "build/layers").exists()
+    }
+}

--- a/src/test/groovy/io/micronaut/gradle/DockerBuildTaskSpec.groovy
+++ b/src/test/groovy/io/micronaut/gradle/DockerBuildTaskSpec.groovy
@@ -4,9 +4,11 @@ import org.gradle.testkit.runner.GradleRunner
 import org.gradle.testkit.runner.TaskOutcome
 import org.junit.Rule
 import org.junit.rules.TemporaryFolder
+import spock.lang.IgnoreIf
 import spock.lang.Specification
 import spock.lang.Unroll
 
+@IgnoreIf({ os.windows })
 class DockerBuildTaskSpec extends Specification {
 
     @Rule TemporaryFolder testProjectDir = new TemporaryFolder()

--- a/src/test/groovy/io/micronaut/gradle/NativeImageMultiProjectSpec.groovy
+++ b/src/test/groovy/io/micronaut/gradle/NativeImageMultiProjectSpec.groovy
@@ -67,7 +67,7 @@ class Application {
         when:
         def result = GradleRunner.create()
                 .withProjectDir(testProjectDir.root)
-                .withArguments('two:nativeImage', '-i')
+                .withArguments('two:nativeImage', '-i', '--stacktrace')
                 .withPluginClasspath()
                 .build()
 

--- a/src/test/groovy/io/micronaut/gradle/NativeImageTaskSpec.groovy
+++ b/src/test/groovy/io/micronaut/gradle/NativeImageTaskSpec.groovy
@@ -111,7 +111,7 @@ class Application {
         when:
         def result = GradleRunner.create()
                 .withProjectDir(testProjectDir.root)
-                .withArguments('nativeImage', '-i')
+                .withArguments('nativeImage', '-i', '--stacktrace')
                 .withPluginClasspath()
                 .build()
 

--- a/src/test/groovy/io/micronaut/gradle/NativeImageTestTaskSpec.groovy
+++ b/src/test/groovy/io/micronaut/gradle/NativeImageTestTaskSpec.groovy
@@ -8,7 +8,7 @@ import org.junit.rules.TemporaryFolder
 import spock.lang.Requires
 import spock.lang.Specification
 
-@Requires({ GraalUtil.isGraalJVM() })
+@Requires({ GraalUtil.isGraalJVM() && !os.windows })
 class NativeImageTestTaskSpec extends Specification {
     @Rule TemporaryFolder testProjectDir = new TemporaryFolder()
 

--- a/src/test/groovy/io/micronaut/gradle/NativeImageTestTaskSpec.groovy
+++ b/src/test/groovy/io/micronaut/gradle/NativeImageTestTaskSpec.groovy
@@ -140,7 +140,7 @@ public class FooControllerTest {
 
         def result = GradleRunner.create()
                 .withProjectDir(testProjectDir.root)
-                .withArguments('testNativeImage')
+                .withArguments('testNativeImage', '--stacktrace')
                 .withPluginClasspath()
                 .build()
 

